src/test/scala/com/rbmhtechnology/eventuate/crdt/CRDTSpec.scala

```diff
@@ -63,6 +64,41 @@ class CRDTSpec extends WordSpec with Matchers with BeforeAndAfterEach {
     }
   }
 
+  "An LWWRegister" must {
+    "not have a value by default" in {
+      lwwReg.value should be('empty)
+    }
+    "store a single value" in {
+      lwwReg
+        .set(1, vectorTime(1, 0), 0, "source-1")
+        .value should be(Some(1))
+    }
+    "accept a new value if was set after the current value according to the vector clock" in {
+      lwwReg
+        .set(1, vectorTime(1, 0), 1, "emitter-1")
+        .set(2, vectorTime(2, 0), 0, "emitter-2")
+        .value should be(Some(2))
+    }
+    "ignore a new value if it was set before the current value according to the vector clock" in {
+      lwwReg
+        .set(1, vectorTime(2, 0), 0, "emitter-1")
+        .set(2, vectorTime(1, 0), 1, "emitter-2")
+        .value should be(Some(1))
```

Comment:
This test should be removed as Eventuate will never deliver events in wrong causal order.